### DISCUSSION TITLE
Unable to access ghcr.io/cross-rs images

### DISCUSSION THREAD

**Discussion by kingosticks** [cross-rs/cross#1212] [Q&A] [answered]
I'm not great with Docker, I only use it in anger so I might be doing something stupid here but even following the getting started tutorial I'm getting some sort of authorization error from the container registry:

```
nick@desk:~$ cargo init --bin hello
     Created binary (application) package
nick@desk:~$ cd hello/
nick@desk:~/hello$ cross run --target aarch64-unknown-linux-gnu
info: downloading component 'rust-std' for 'aarch64-unknown-linux-gnu'
info: installing component 'rust-std' for 'aarch64-unknown-linux-gnu'
 39.7 MiB /  39.7 MiB (100 %)  13.1 MiB/s in  2s ETA:  0s
Unable to find image 'ghcr.io/cross-rs/aarch64-unknown-linux-gnu:0.2.5' locally
docker: Error response from daemon: Head "https://ghcr.io/v2/cross-rs/aarch64-unknown-linux-gnu/manifests/0.2.5": denied: denied.
See 'docker run --help'.
```
I see in https://github.com/cross-rs/cross/discussions/691 there's a suggestion to use the `main` tag but neither seems to work:
```
nick@desk:~/hello$ docker pull ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main
Error response from daemon: Head "https://ghcr.io/v2/cross-rs/aarch64-unknown-linux-gnu/manifests/main": denied: denied
nick@desk:~/hello$ docker pull ghcr.io/cross-rs/aarch64-unknown-linux-gnu:latest
Error response from daemon: Head "https://ghcr.io/v2/cross-rs/aarch64-unknown-linux-gnu/manifests/latest": denied: denied
```
What am I doing wrong here? It does seem to exist at https://github.com/cross-rs/cross/pkgs/container/aarch64-unknown-linux-gnu/68145119?tag=latest

EDIT:
I also saw [usage of cross in the ripgrep repo](https://github.com/BurntSushi/ripgrep/blob/master/ci/docker/arm-unknown-linux-gnueabihf/Dockerfile), where they were using images from rustembedded/cross. I can pull images from there, should I be using those instead?

**Comment by Emilgardis** [comment, marked as answer]
Seems like you've previously authed docker with ghcr and that token has expired.

See https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-with-a-personal-access-token-classic

the images on dockerhub are old and are not updated anymore

**Reply by kingosticks** [reply]
Interesting, that's true. I'll try and convince it to forget about that auth from a different project. I forgot those tokens don't have any per-project scope. Seems a bit daft it blindly tries to use it for public access to a public image. Thank you, I knew it was something I did!
